osu.Desktop/OsuGameDesktop.cs

```diff
@@ -0,0 +1,60 @@
+﻿using osu.Game;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using osu.Framework.Platform;
+using osu.Framework.Desktop.Platform;
+using osu.Game.Database;
+
+namespace osu.Desktop
+{
+    class OsuGameDesktop : OsuGame
+    {
+        public OsuGameDesktop(string[] args = null)
+            : base(args)
+        {
+
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+        }
+
+        public override void SetHost(BasicGameHost host)
+        {
+            base.SetHost(host);
+            var desktopWindow = host.Window as DesktopGameWindow;
+            if(desktopWindow != null)
+            {
+                desktopWindow.DragEnter += dragEnter;
+                desktopWindow.DragDrop += dragDrop;
+            }
+        }
+
+        private void dragDrop(DragEventArgs e)
+        {
+            // this method will only be executed if e.Effect in dragEnter gets set to something other that None.
+            var dropData = e.Data.GetData(DataFormats.FileDrop) as Array;
+            var filePaths = dropData.Cast<object>().Select(f => f.ToString()).ToArray();
+            Schedule(delegate { Dependencies.Get<BeatmapDatabase>().Import(filePaths); });

```

Comment:
as there are now two usages of this, I'd prefer you make an `Import(string params paths)` method in OsuGame or OsuGameBase and move this code there (including the schedule).